src/amd/compiler/aco_lower_bool_phis.cpp

```diff
@@ -122,22 +123,24 @@ Temp write_ssa(Program *program, Block *block, ssa_state *state, unsigned previo
 
 void insert_before_branch(Block *block, aco_ptr<Instruction> instr)
 {
-   int end = block->instructions.size() - 1;
-   if (block->instructions[end]->format == Format::PSEUDO_BRANCH)
-      block->instructions.emplace(std::prev(block->instructions.end()), std::move(instr));
+   if (block->instructions.back()->format == Format::PSEUDO_BRANCH)
```

Comment:
Looking at this code now, I don't think we can ever have a case where a block doesn't end with a PSEUDO_BRANCH instruction and this condition is always true. This seems to be the case with every shader in our pipeline-db. 

So I think this function could be simplified to:
```c++
void insert_before_branch(Block *block, aco_ptr<Instruction> instr)
{
   block->instructions.insert(std::prev(block->instructions.end()), std::move(instr));
}
```

Reply:
That is great.

I will add an assert just to be save